File: experiments/llm-reasoning-cost/generated_code/java/07_word_freq_codex.java
import java.util.*;

public class Main {
    public static Map<String, Integer> count_words(String text) {
        Map<String, Integer> counts = new HashMap<>();
        String[] words = text.toLowerCase().trim().split("\\s+");

        for (String word : words) {
            if (!word.isEmpty()) {
                counts.put(word, counts.getOrDefault(word, 0) + 1);
            }
        }

        return counts;
    }

    public static void print_top(Map<String, Integer> wordCounts, int n) {
        List<Map.Entry<String, Integer>> entries = new ArrayList<>(wordCounts.entrySet());

        entries.sort((a, b) -> {
            int cmp = Integer.compare(b.getValue(), a.getValue()); // count desc
            if (cmp != 0) return cmp;
            return a.getKey().compareTo(b.getKey()); // word asc
        });

        int limit = Math.min(n, entries.size());
        for (int i = 0; i < limit; i++) {
            Map.Entry<String, Integer> e = entries.get(i);
            System.out.println("  " + e.getValue() + " " + e.getKey());
        }
    }

    public static void main(String[] args) {
        String text = "the cat sat on the mat the cat sat";
        Map<String, Integer> counts = count_words(text);
        print_top(counts, 5);
    }
}

File: experiments/llm-reasoning-cost/generated_code/go/07_word_freq_codex.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func count_words(text string) map[string]int32 {
	counts := make(map[string]int32)
	words := strings.Fields(strings.ToLower(text))
	for _, w := range words {
		counts[w]++
	}
	return counts
}

func print_top(counts map[string]int32, n int32) {
	type entry struct {
		word  string
		count int32
	}

	entries := make([]entry, 0, len(counts))
	for w, c := range counts {
		entries = append(entries, entry{word: w, count: c})
	}

	sort.Slice(entries, func(i, j int) bool {
		if entries[i].count != entries[j].count {
			return entries[i].count > entries[j].count
		}
		return entries[i].word < entries[j].word
	})

	limit := int(n)
	if limit > len(entries) {
		limit = len(entries)
	}
	for i := 0; i < limit; i++ {
		fmt.Printf("  %d %s\n", entries[i].count, entries[i].word)
	}
}

func main() {
	text := "the cat sat on the mat the cat sat"
	counts := count_words(text)
	print_top(counts, 5)
}
